[Bug]: Milvus panic in ci test with error `panic in a function that cannot unwind`

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Environment

```markdown
- Milvus version:master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior



```
2025-11-14 16:49:03.114	
[2025-11-14T08:49:03Z INFO  tantivy::directory::managed_directory] Garbage collect
2025-11-14 16:49:06.109	

2025-11-14 16:49:06.109	
thread '<unnamed>' panicked at src/index_reader_c.rs:20:5:

2025-11-14 16:49:06.109	
assertion failed: tantivy_index_exist(path)
2025-11-14 16:49:06.109	
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2025-11-14 16:49:06.109	

2025-11-14 16:49:06.109	
thread '<unnamed>' panicked at library/core/src/panicking.rs:225:5:
2025-11-14 16:49:06.109	
panic in a function that cannot unwind
2025-11-14 16:49:06.109	
stack backtrace:
2025-11-14 16:49:06.110	
[2025-11-14T08:49:06Z INFO  tantivy_binding::index_writer_v7::index_writer] create index writer, field_name: 103, data_type: I64, tantivy_index_version 7
2025-11-14 16:49:06.110	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::segment_updater] save metas
2025-11-14 16:49:06.426	
   0:     0x7f789960f672 - std::backtrace_rs::backtrace::libunwind::trace::h9c1aa7b29a521839
2025-11-14 16:49:06.426	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
2025-11-14 16:49:06.426	
   1:     0x7f789960f672 - std::backtrace_rs::backtrace::trace_unsynchronized::hb123c31478ec901c
2025-11-14 16:49:06.426	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
2025-11-14 16:49:06.426	
   2:     0x7f789960f672 - std::sys::backtrace::_print_fmt::hdda75a118fd2034a
2025-11-14 16:49:06.426	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/sys/backtrace.rs:66:9
2025-11-14 16:49:06.426	
   3:     0x7f789960f672 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::hf435e8e9347709a8
2025-11-14 16:49:06.426	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/sys/backtrace.rs:39:26
2025-11-14 16:49:06.429	
   4:     0x7f789965e033 - core::fmt::rt::Argument::fmt::h9802ea71fd88c728
2025-11-14 16:49:06.429	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/fmt/rt.rs:173:76
2025-11-14 16:49:06.430	
   5:     0x7f789965e033 - core::fmt::write::h0a51fad3804c5e7c
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/fmt/mod.rs:1465:25
2025-11-14 16:49:06.430	
   6:     0x7f7899603c33 - std::io::default_write_fmt::h7e00b0a8732ee2a2
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/io/mod.rs:639:11
2025-11-14 16:49:06.430	
   7:     0x7f7899603c33 - std::io::Write::write_fmt::h9759e4151bf4a45e
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/io/mod.rs:1954:13
2025-11-14 16:49:06.430	
   8:     0x7f789960f4c2 - std::sys::backtrace::BacktraceLock::print::h1ec5ce5bb8ee285e
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/sys/backtrace.rs:42:9
2025-11-14 16:49:06.430	
   9:     0x7f789961309c - std::panicking::default_hook::{{closure}}::h5ffefe997a3c75e4
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/panicking.rs:300:27
2025-11-14 16:49:06.430	
  10:     0x7f7899612e9f - std::panicking::default_hook::h820c77ba0601d6bb
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/panicking.rs:327:9
2025-11-14 16:49:06.430	
  11:     0x7f7899613aa2 - std::panicking::rust_panic_with_hook::h8b29cbe181d50030
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/panicking.rs:833:13
2025-11-14 16:49:06.430	
  12:     0x7f7899613826 - std::panicking::begin_panic_handler::{{closure}}::h9f5b6f6dc6fde83e
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/panicking.rs:699:13
2025-11-14 16:49:06.430	
  13:     0x7f789960fb79 - std::sys::backtrace::__rust_end_short_backtrace::hd7b0c344383b0b61
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/sys/backtrace.rs:168:18
2025-11-14 16:49:06.430	
  14:     0x7f78996134ed - __rustc[5224e6b81cd82a8f]::rust_begin_unwind
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/std/src/panicking.rs:697:5
2025-11-14 16:49:06.430	
  15:     0x7f789469a18d - core::panicking::panic_nounwind_fmt::runtime::h2095774d07fc8a63
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/panicking.rs:117:22
2025-11-14 16:49:06.430	
  16:     0x7f789469a18d - core::panicking::panic_nounwind_fmt::hdcc420bb84b07b49
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/intrinsics/mod.rs:2340:9
2025-11-14 16:49:06.430	
  17:     0x7f789469a222 - core::panicking::panic_nounwind::h681d69c581350003
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/panicking.rs:225:5
2025-11-14 16:49:06.430	
  18:     0x7f789469a3e2 - core::panicking::panic_cannot_unwind::h4320389209385803
2025-11-14 16:49:06.430	
                               at /rustc/29483883eed69d5fb4db01964cdf2af4d86e9cb2/library/core/src/panicking.rs:330:5
2025-11-14 16:49:06.460	
  19:     0x7f7898721439 - tantivy_load_index
2025-11-14 16:49:06.460	
                               at /workspace/source/internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_c.rs:15:1
2025-11-14 16:49:06.491	
  20:     0x7f789582e165 - _ZN6milvus7tantivy19TantivyIndexWrapperC4EPKcbPFvPvPKjmE
2025-11-14 16:49:06.491	
                               at /workspace/source/internal/core/thirdparty/tantivy/tantivy-wrapper.h:124:64
2025-11-14 16:49:06.491	
  21:     0x7f789582e165 - _ZSt10_ConstructIN6milvus7tantivy19TantivyIndexWrapperEJPKcRbRFvPvPKjmEEEvPT_DpOT0_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/stl_construct.h:119:7
2025-11-14 16:49:06.491	
  22:     0x7f789582e165 - _ZNSt16allocator_traitsISaIvEE9constructIN6milvus7tantivy19TantivyIndexWrapperEJPKcRbRFvPvPKjmEEEEvRS0_PT_DpOT0_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/alloc_traits.h:635:19
2025-11-14 16:49:06.491	
  23:     0x7f789582e165 - _ZNSt23_Sp_counted_ptr_inplaceIN6milvus7tantivy19TantivyIndexWrapperESaIvELN9__gnu_cxx12_Lock_policyE2EEC4IJPKcRbRFvPvPKjmEEEES3_DpOT_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/shared_ptr_base.h:604:39
2025-11-14 16:49:06.491	
  24:     0x7f789582e165 - _ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC4IN6milvus7tantivy19TantivyIndexWrapperESaIvEJPKcRbRFvPvPKjmEEEERPT_St20_Sp_alloc_shared_tagIT0_EDpOT1_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/shared_ptr_base.h:971:16
2025-11-14 16:49:06.491	
  25:     0x7f789582e165 - _ZNSt12__shared_ptrIN6milvus7tantivy19TantivyIndexWrapperELN9__gnu_cxx12_Lock_policyE2EEC4ISaIvEJPKcRbRFvPvPKjmEEEESt20_Sp_alloc_shared_tagIT_EDpOT0_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/shared_ptr_base.h:1712:14
2025-11-14 16:49:06.491	
  26:     0x7f789582e165 - _ZNSt10shared_ptrIN6milvus7tantivy19TantivyIndexWrapperEEC4ISaIvEJPKcRbRFvPvPKjmEEEESt20_Sp_alloc_shared_tagIT_EDpOT0_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/shared_ptr.h:464:59
2025-11-14 16:49:06.491	
  27:     0x7f789582e165 - _ZSt11make_sharedIN6milvus7tantivy19TantivyIndexWrapperEJPKcRbRFvPvPKjmEEESt10shared_ptrINSt9enable_ifIXntsrSt8is_arrayIT_E5valueESE_E4typeEEDpOT0_
2025-11-14 16:49:06.491	
                               at /usr/include/c++/12/bits/shared_ptr.h:1010:39
2025-11-14 16:49:06.491	
  28:     0x7f789582e165 - _ZN6milvus5index20InvertedIndexTantivyINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE4LoadENS_6tracer12TraceContextERKN8nlohmann16json_abi_v3_11_310basic_jsonISt3mapSt6vectorS7_blmdSaNSC_14adl_serializerESF_IhSaIhEEvEE
2025-11-14 16:49:06.492	
                               at /workspace/source/internal/core/src/index/InvertedIndexTantivy.cpp:198:14
2025-11-14 16:49:06.514	
  29:     0x7f78956a92cb - _ZN6milvus5index17HybridScalarIndexINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE4LoadENS_6tracer12TraceContextERKN8nlohmann16json_abi_v3_11_310basic_jsonISt3mapSt6vectorS7_blmdSaNSC_14adl_serializerESF_IhSaIhEEvEE
2025-11-14 16:49:06.514	
                               at /workspace/source/internal/core/src/index/HybridScalarIndex.cpp:382:16
2025-11-14 16:49:06.529	
  30:     0x7f7896ae1e7a - _ZN6milvus7segcore19storagev1translator21SealedIndexTranslator9get_cellsERKSt6vectorIlSaIlEE
2025-11-14 16:49:06.529	
                               at /workspace/source/internal/core/src/segcore/storagev1translator/SealedIndexTranslator.cpp:136:16
2025-11-14 16:49:06.549	
  31:     0x7f789624e01a - _ZZN6milvus12cachinglayer9CacheSlotINS_5index9IndexBaseEE7RunLoadEOSt13unordered_setIlSt4hashIlESt8equal_toIlESaIlEENSt6chrono8durationIlSt5ratioILl1ELl1000EEEEENKUlvE_clEv
2025-11-14 16:49:06.549	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/CacheSlot.h:368:22
2025-11-14 16:49:06.549	
  32:     0x7f7896251767 - _ZN6milvus12cachinglayer9CacheSlotINS_5index9IndexBaseEE7RunLoadEOSt13unordered_setIlSt4hashIlESt8equal_toIlESaIlEENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE
2025-11-14 16:49:06.549	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/CacheSlot.h:378:34
2025-11-14 16:49:06.550	
  33:     0x7f789625856a - _ZN6milvus12cachinglayer9CacheSlotINS_5index9IndexBaseEE11PinInternalEPNS_9OpContextERKSt6vectorIlSaIlEENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE
2025-11-14 16:49:06.550	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/CacheSlot.h:315:20
2025-11-14 16:49:06.614	
  34:     0x7f789644d4b7 - _ZN6milvus12cachinglayer9CacheSlotINS_5index9IndexBaseEE14PinCellsDirectEPNS_9OpContextERKSt6vectorIlSaIlEENSt6chrono8durationIlSt5ratioILl1ELl1000EEEE
2025-11-14 16:49:06.614	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/CacheSlot.h:236:59
2025-11-14 16:49:06.636	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::index_writer] Preparing commit
2025-11-14 16:49:06.641	
  35:     0x7f789684a8e5 - _ZN6milvus12cachinglayer9CacheSlotINS_5index9IndexBaseEE6WarmupEv
2025-11-14 16:49:06.641	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/CacheSlot.h:93:9
2025-11-14 16:49:06.641	
  36:     0x7f789684a8e5 - _ZN6milvus12cachinglayer7Manager15CreateCacheSlotINS_5index9IndexBaseEEESt10shared_ptrINS0_9CacheSlotIT_EEESt10unique_ptrINS0_10TranslatorIS7_EESt14default_deleteISC_EE
2025-11-14 16:49:06.641	
                               at /workspace/source/cmake_build/thirdparty/milvus-common/milvus-common-src/include/cachinglayer/Manager.h:50:27
2025-11-14 16:49:06.641	
  37:     0x7f789684a8e5 - _ZN6milvus7segcore13LoadIndexDataERNS_6tracer12TraceContextEPNS0_13LoadIndexInfoE
2025-11-14 16:49:06.641	
                               at /workspace/source/internal/core/src/segcore/Utils.cpp:1314:69
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::index_writer] Prepared commit 2001
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::prepared_commit] committing 2001
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::segment_updater] save metas
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::segment_updater] Running garbage collection
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::directory::managed_directory] Garbage collect
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::indexer::segment_updater] Running garbage collection
2025-11-14 16:49:06.655	
[2025-11-14T08:49:06Z INFO  tantivy::directory::managed_directory] Garbage collect
2025-11-14 16:49:06.661	
  38:     0x7f78968930f2 - AppendIndexV2
2025-11-14 16:49:06.661	
                               at /workspace/source/internal/core/src/segcore/load_index_c.cpp:249:22
2025-11-14 16:49:07.478	
  39:          0x6c1e06c - _cgo_ec19d29abd14_Cfunc_AppendIndexV2
2025-11-14 16:49:07.478	
                               at /tmp/go-build/cgo-gcc-prolog:135:11
2025-11-14 16:49:07.478	
  40:          0x28906c4 - runtime.asmcgocall
2025-11-14 16:49:07.478	
                               at /usr/local/go/src/runtime/asm_amd64.s:923:0
2025-11-14 16:49:07.478	
thread caused non-unwinding panic. aborting.
2025-11-14 16:49:07.481	
SIGABRT: abort
2025-11-14 16:49:07.481	
PC=0x7f7891feb9fc m=1825 sigcode=18446744073709551610
2025-11-14 16:49:07.481	
signal arrived during cgo execution
2025-11-14 16:49:07.481	

2025-11-14 16:49:07.481	
goroutine 2796839 gp=0x10c0031e61c0 m=1825 mp=0x10c001282008 [syscall, locked to thread]:
2025-11-14 16:49:08.863	
non-Go function
2025-11-14 16:49:08.863	
	pc=0x7f7891feb9fc
2025-11-14 16:49:08.863	
non-Go function
2025-11-14 16:49:08.863	
	pc=0x7f7891f97475
2025-11-14 16:49:08.863	
non-Go function
2025-11-14 16:49:08.863	
	pc=0x7f7891f7d7f2
2025-11-14 16:49:08.902	
_ZN3std3sys3pal4unix14abort_internal17hc4415ac680839f8aE
2025-11-14 16:49:08.902	
	library/std/src/sys/pal/unix/mod.rs:366 pc=0x7f78946986e9
2025-11-14 16:49:08.902	
_ZN3std9panicking20rust_panic_with_hook17h8b29cbe181d50030E
2025-11-14 16:49:08.902	
	library/std/src/panicking.rs:0 pc=0x7f7899613c09
2025-11-14 16:49:08.902	
_ZN3std9panicking19begin_panic_handler28_$u7b$$u7b$closure$u7d$$u7d$17h9f5b6f6dc6fde83eE
2025-11-14 16:49:08.902	
	library/std/src/panicking.rs:699 pc=0x7f7899613825
2025-11-14 16:49:08.902	
_ZN3std3sys9backtrace26__rust_end_short_backtrace17hd7b0c344383b0b61E
2025-11-14 16:49:08.902	
	library/std/src/sys/backtrace.rs:168 pc=0x7f789960fb78
2025-11-14 16:49:08.902	
_RNvCs73fAdSrgOJL_7___rustc17rust_begin_unwind
2025-11-14 16:49:08.902	
	library/std/src/panicking.rs:697 pc=0x7f78996134ec
2025-11-14 16:49:08.910	
_ZN4core9panicking18panic_nounwind_fmt7runtime17h2095774d07fc8a63E
2025-11-14 16:49:08.910	
	library/core/src/panicking.rs:117 pc=0x7f789469a18c
2025-11-14 16:49:08.910	
_ZN4core9panicking18panic_nounwind_fmt17hdcc420bb84b07b49E
2025-11-14 16:49:08.910	
	library/core/src/intrinsics/mod.rs:2340 pc=0x7f789469a18c
2025-11-14 16:49:08.910	
_ZN4core9panicking14panic_nounwind17h681d69c581350003E
2025-11-14 16:49:08.910	
	library/core/src/panicking.rs:225 pc=0x7f789469a221
2025-11-14 16:49:08.910	
runtime.cgocall(0x6c1df90, 0x10c00aa4c648)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/cgocall.go:167 +0x4b fp=0x10c00aa4c620 sp=0x10c00aa4c5e8 pc=0x288394b
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/internal/querynodev2/segments._Cfunc_AppendIndexV2({0x10c011bdced0, 0x10c011bdcee0, 0x0, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}}, ...)
2025-11-14 16:49:08.910	
	_cgo_gotypes.go:192 +0x4f fp=0x10c00aa4c648 sp=0x10c00aa4c620 pc=0x673d0ef
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/internal/querynodev2/segments.(*LoadIndexInfo).loadIndex.func1.2(...)
2025-11-14 16:49:08.910	
	/workspace/source/internal/querynodev2/segments/load_index_info.go:167
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/internal/querynodev2/segments.(*LoadIndexInfo).loadIndex.func1()
2025-11-14 16:49:08.910	
	/workspace/source/internal/querynodev2/segments/load_index_info.go:167 +0x115 fp=0x10c00aa4c720 sp=0x10c00aa4c648 pc=0x673fed5
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/pkg/v2/util/conc.(*Pool[...]).Submit.func1()
2025-11-14 16:49:08.910	
	/workspace/source/pkg/util/conc/pool.go:82 +0x9e fp=0x10c00aa4c780 sp=0x10c00aa4c720 pc=0x5911ede
2025-11-14 16:49:08.910	
github.com/panjf2000/ants/v2.(*goWorker).run.func1()
2025-11-14 16:49:08.910	
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/worker.go:73 +0x8a fp=0x10c00aa4c7e0 sp=0x10c00aa4c780 pc=0x51e30aa
2025-11-14 16:49:08.910	
runtime.goexit({})
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c00aa4c7e8 sp=0x10c00aa4c7e0 pc=0x2890a41
2025-11-14 16:49:08.910	
created by github.com/panjf2000/ants/v2.(*goWorker).run in goroutine 2796858
2025-11-14 16:49:08.910	
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/worker.go:50 +0x65
2025-11-14 16:49:08.910	

2025-11-14 16:49:08.910	
goroutine 1 gp=0x10c000002380 m=nil [chan receive, 53 minutes]:
2025-11-14 16:49:08.910	
runtime.gopark(0x9d34a7eb9c116d?, 0x7f761c240a28?, 0x78?, 0xfa?, 0x18?)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c001fa95d0 sp=0x10c001fa95b0 pc=0x2887dce
2025-11-14 16:49:08.910	
runtime.chanrecv(0x10c0013055e0, 0x0, 0x1)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/chan.go:664 +0x445 fp=0x10c001fa9648 sp=0x10c001fa95d0 pc=0x281b025
2025-11-14 16:49:08.910	
runtime.chanrecv1(0xbb4e090?, 0x7b72b80?)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/chan.go:506 +0x12 fp=0x10c001fa9670 sp=0x10c001fa9648 pc=0x281abb2
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0x10c001ebc320)
2025-11-14 16:49:08.910	
	/workspace/source/cmd/roles/roles.go:554 +0x15a5 fp=0x10c001fa9ba8 sp=0x10c001fa9670 pc=0x6885645
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xbbb10a0, {0x10c000072990, 0x3, 0x3}, 0x10c000aecee0)
2025-11-14 16:49:08.910	
	/workspace/source/cmd/milvus/run.go:48 +0x2ae fp=0x10c001fa9c68 sp=0x10c001fa9ba8 pc=0x689264e
2025-11-14 16:49:08.910	
github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0x10c000072990, 0x3, 0x3})
2025-11-14 16:49:08.910	
	/workspace/source/cmd/milvus/milvus.go:60 +0x1e2 fp=0x10c001fa9cd8 sp=0x10c001fa9c68 pc=0x6892302
2025-11-14 16:49:08.910	
main.main()
2025-11-14 16:49:08.910	
	/workspace/source/cmd/main.go:102 +0x285 fp=0x10c001fa9f50 sp=0x10c001fa9cd8 pc=0x6896a45
2025-11-14 16:49:08.910	
runtime.main()
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:283 +0x28b fp=0x10c001fa9fe0 sp=0x10c001fa9f50 pc=0x285058b
2025-11-14 16:49:08.910	
runtime.goexit({})
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c001fa9fe8 sp=0x10c001fa9fe0 pc=0x2890a41
2025-11-14 16:49:08.910	

2025-11-14 16:49:08.910	
goroutine 17 gp=0x10c0000028c0 m=1 mp=0x10c000156008 [syscall, locked to thread]:
2025-11-14 16:49:08.910	
runtime.goexit({})
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c000164fe8 sp=0x10c000164fe0 pc=0x2890a41
2025-11-14 16:49:08.910	

2025-11-14 16:49:08.910	
goroutine 2 gp=0x10c000002e00 m=nil [force gc (idle), 53 minutes]:
2025-11-14 16:49:08.910	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000150fa8 sp=0x10c000150f88 pc=0x2887dce
2025-11-14 16:49:08.910	
runtime.goparkunlock(...)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:441
2025-11-14 16:49:08.910	
runtime.forcegchelper()
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:348 +0xb3 fp=0x10c000150fe0 sp=0x10c000150fa8 pc=0x28508d3
2025-11-14 16:49:08.910	
runtime.goexit({})
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c000150fe8 sp=0x10c000150fe0 pc=0x2890a41
2025-11-14 16:49:08.910	
created by runtime.init.7 in goroutine 1
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:336 +0x1a
2025-11-14 16:49:08.910	

2025-11-14 16:49:08.910	
goroutine 3 gp=0x10c000003340 m=nil [GC sweep wait]:
2025-11-14 16:49:08.910	
runtime.gopark(0xbb43e01?, 0xbb43ec0?, 0xc?, 0x9?, 0x1?)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000151780 sp=0x10c000151760 pc=0x2887dce
2025-11-14 16:49:08.910	
runtime.goparkunlock(...)
2025-11-14 16:49:08.910	
	/usr/local/go/src/runtime/proc.go:441
2025-11-14 16:49:08.911	
runtime.bgsweep(0x10c00014a080)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgcsweep.go:316 +0xdf fp=0x10c0001517c8 sp=0x10c000151780 pc=0x2836bbf
2025-11-14 16:49:08.911	
runtime.gcenable.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:204 +0x25 fp=0x10c0001517e0 sp=0x10c0001517c8 pc=0x282ada5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c0001517e8 sp=0x10c0001517e0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcenable in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:204 +0x66
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 4 gp=0x10c000003500 m=nil [sleep]:
2025-11-14 16:49:08.911	
runtime.gopark(0x10c00017e060?, 0x3659015a6fc775?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000151f38 sp=0x10c000151f18 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.goparkunlock(...)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:441
2025-11-14 16:49:08.911	
runtime.(*scavengerState).sleep(0xbb4fd80, 0x412e91b400000000)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgcscavenge.go:504 +0xf6 fp=0x10c000151fa8 sp=0x10c000151f38 pc=0x2834636
2025-11-14 16:49:08.911	
runtime.bgscavenge(0x10c00014a080)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgcscavenge.go:662 +0x74 fp=0x10c000151fc8 sp=0x10c000151fa8 pc=0x2834a34
2025-11-14 16:49:08.911	
runtime.gcenable.gowrap2()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:205 +0x25 fp=0x10c000151fe0 sp=0x10c000151fc8 pc=0x282ad45
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c000151fe8 sp=0x10c000151fe0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcenable in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:205 +0xa5
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 5 gp=0x10c000003dc0 m=nil [finalizer wait]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x7e98478?, 0x0?, 0x20?, 0x1000000010?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c00a39fe30 sp=0x10c00a39fe10 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.runfinq()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mfinal.go:196 +0x107 fp=0x10c00a39ffe0 sp=0x10c00a39fe30 pc=0x2829ce7
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c00a39ffe8 sp=0x10c00a39ffe0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.createfing in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mfinal.go:166 +0x3d
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 6 gp=0x10c0002de000 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000152738 sp=0x10c000152718 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c0001527c8 sp=0x10c000152738 pc=0x282d209
2025-11-14 16:49:08.911	
runtime.gcBgMarkStartWorkers.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x25 fp=0x10c0001527e0 sp=0x10c0001527c8 pc=0x282d0e5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c0001527e8 sp=0x10c0001527e0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcBgMarkStartWorkers in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x105
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 18 gp=0x10c000202380 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c00014c738 sp=0x10c00014c718 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c00014c7c8 sp=0x10c00014c738 pc=0x282d209
2025-11-14 16:49:08.911	
runtime.gcBgMarkStartWorkers.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x25 fp=0x10c00014c7e0 sp=0x10c00014c7c8 pc=0x282d0e5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c00014c7e8 sp=0x10c00014c7e0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcBgMarkStartWorkers in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x105
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 34 gp=0x10c0009c0000 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c0009c6738 sp=0x10c0009c6718 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c0009c67c8 sp=0x10c0009c6738 pc=0x282d209
2025-11-14 16:49:08.911	
runtime.gcBgMarkStartWorkers.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x25 fp=0x10c0009c67e0 sp=0x10c0009c67c8 pc=0x282d0e5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c0009c67e8 sp=0x10c0009c67e0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcBgMarkStartWorkers in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x105
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 35 gp=0x10c0009c01c0 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c0009c6f38 sp=0x10c0009c6f18 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c0009c6fc8 sp=0x10c0009c6f38 pc=0x282d209
2025-11-14 16:49:08.911	
runtime.gcBgMarkStartWorkers.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x25 fp=0x10c0009c6fe0 sp=0x10c0009c6fc8 pc=0x282d0e5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c0009c6fe8 sp=0x10c0009c6fe0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcBgMarkStartWorkers in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x105
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 7 gp=0x10c0002de1c0 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000152f38 sp=0x10c000152f18 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c000152fc8 sp=0x10c000152f38 pc=0x282d209
2025-11-14 16:49:08.911	
runtime.gcBgMarkStartWorkers.gowrap1()
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x25 fp=0x10c000152fe0 sp=0x10c000152fc8 pc=0x282d0e5
2025-11-14 16:49:08.911	
runtime.goexit({})
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0x10c000152fe8 sp=0x10c000152fe0 pc=0x2890a41
2025-11-14 16:49:08.911	
created by runtime.gcBgMarkStartWorkers in goroutine 1
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1339 +0x105
2025-11-14 16:49:08.911	

2025-11-14 16:49:08.911	
goroutine 8 gp=0x10c0002de380 m=nil [GC worker (idle), 53 minutes]:
2025-11-14 16:49:08.911	
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/proc.go:435 +0xce fp=0x10c000153738 sp=0x10c000153718 pc=0x2887dce
2025-11-14 16:49:08.911	
runtime.gcBgMarkWorker(0x10c000188150)
2025-11-14 16:49:08.911	
	/usr/local/go/src/runtime/mgc.go:1423 +0xe9 fp=0x10c0001537c8 sp=0x10c000153738 pc=0x282d209
```

### Expected Behavior

_No response_

### Steps To Reproduce

```markdown

```

### Milvus Log

failed ci job: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-45574/1/pipeline

```

2025-11-14T08:50:49Z {container="step-test"} FAILED milvus_client/test_milvus_client_search.py::TestMilvusClientSearchNullExpr::test_milvus_client_search_null_expr_array[IS NULL-INVERTED-True-False-True-False]

2025-11-14T08:50:49Z {container="step-test"} !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!

2025-11-14T08:50:49Z {container="step-test"} !!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!

2025-11-14T08:50:49Z {container="step-test"} = 1 failed, 1283 passed, 74 skipped, 13 warnings, 7 rerun in 1887.33s (0:31:27) =

2025-11-14T08:50:54Z {container="prepare"} 2025/11/14 08:50:54 Entrypoint initialization

2025-11-14T08:50:56Z {container="place-scripts"} 2025/11/14 08:50:56 Decoded script /tekton/scripts/script-0-2vgb6

2025-11-14T08:50:56Z {container="place-scripts"} 2025/11/14 08:50:56 Decoded script /tekton/scripts/script-1-zl42j

2025-11-14T08:50:56Z {container="place-scripts"} 2025/11/14 08:50:56 Decoded script /tekton/scripts/script-2-m8r5c

2025-11-14T08:51:00Z {container="step-check-status"} ----------------Pod Status --------------------------------------------

2025-11-14T08:51:00Z {container="step-check-status"} ms-45574-1-py-pr-etcd-0                                    1/1     Running            0                 56m     10.104.31.35    4am-node34   <none>           <none>

2025-11-14T08:51:00Z {container="step-check-status"} ms-45574-1-py-pr-milvus-standalone-875cbc484-fbgxv         1/1     Running            1 (101s ago)      56m     10.104.31.39    4am-node34   <none>           <none>

2025-11-14T08:51:00Z {container="step-check-status"} ms-45574-1-py-pr-minio-54db76855b-z277m                    1/1     Running            0                 56m     10.104.31.34    4am-node34   <none>           <none>
```


pod `ms-45574-1-py-pr-milvus-standalone-875cbc484-fbgxv ` restarted during test 


### Anything else?

_No response_

--- a/internal/core/src/index/TextMatchIndex.cpp
+++ b/internal/core/src/index/TextMatchIndex.cpp
@@ -36,7 +36,8 @@ StripTextLogPrefix(const std::string& path, const std::string& base_prefix) {
 TextMatchIndex::TextMatchIndex(int64_t commit_interval_in_ms,
                                const char* unique_id,
                                const char* analyzer_name,
-                               const char* analyzer_params)
+                               const char* analyzer_params,
+                               bool enable_background_merge)
     : commit_interval_in_ms_(commit_interval_in_ms),
       last_commit_time_(stdclock::now()) {
     d_type_ = TantivyDataType::Text;
@@ -47,7 +48,11 @@ TextMatchIndex::TextMatchIndex(int64_t commit_interval_in_ms,
         TANTIVY_INDEX_LATEST_VERSION /* Growing segment has no reason to use old version index*/
         ,
         analyzer_name,
-        analyzer_params);
+        analyzer_params,
+        /*analyzer_extra_info=*/"",
+        milvus::tantivy::DEFAULT_NUM_THREADS,
+        milvus::tantivy::DEFAULT_OVERALL_MEMORY_BUDGET_IN_BYTES,
+        enable_background_merge);
     set_is_growing(true);
 }
 

--- a/internal/core/src/index/TextMatchIndex.h
+++ b/internal/core/src/index/TextMatchIndex.h
@@ -24,10 +24,16 @@ using stdclock = std::chrono::high_resolution_clock;
 class TextMatchIndex : public InvertedIndexTantivy<std::string> {
  public:
     // for growing segment.
+    // In-memory writer. enable_background_merge must be true only for a
+    // long-lived growing segment (periodic commits would otherwise grow the
+    // segment count unbounded); a sealed interim index passes false so that
+    // finish()'s explicit merge-all is the only merge and cannot race a
+    // background policy merge.
     explicit TextMatchIndex(int64_t commit_interval_in_ms,
                             const char* unique_id,
                             const char* analyzer_name,
-                            const char* analyzer_params);
+                            const char* analyzer_params,
+                            bool enable_background_merge);
     // for sealed segment to create index from raw data during loading.
     explicit TextMatchIndex(const std::string& path,
                             const char* unique_id,

--- a/internal/core/src/index/TextMatchIndexTest.cpp
+++ b/internal/core/src/index/TextMatchIndexTest.cpp
@@ -245,7 +245,8 @@ TEST(TextMatch, Index) {
     auto index = std::make_unique<Index>(std::numeric_limits<int64_t>::max(),
                                          "unique_id",
                                          "milvus_tokenizer",
-                                         "{}");
+                                         "{}",
+                                         /*enable_background_merge=*/false);
     index->CreateReader(milvus::index::SetBitsetSealed);
     index->AddTextSealed("football, basketball, pingpang", true, 0);
     index->AddTextSealed("", false, 1);
@@ -381,8 +382,11 @@ TEST(TextMatch, BuildIndexFromFieldDataMultiBatchNullable) {
 
     std::vector<milvus::FieldDataPtr> field_datas = {batch0, batch1, batch2};
 
-    auto index = std::make_unique<Index>(
-        200, "test_multi_batch", "milvus_tokenizer", "{}");
+    auto index = std::make_unique<Index>(200,
+                                         "test_multi_batch",
+                                         "milvus_tokenizer",
+                                         "{}",
+                                         /*enable_background_merge=*/true);
     index->CreateReader(milvus::index::SetBitsetGrowing);
     index->RegisterAnalyzer("milvus_tokenizer", "{}");
 
@@ -708,8 +712,11 @@ TEST(TextMatch, BuildIndexFromFieldDataSingleBatchNullable) {
 
     std::vector<milvus::FieldDataPtr> field_datas = {fd};
 
-    auto index = std::make_unique<Index>(
-        200, "test_single_batch", "milvus_tokenizer", "{}");
+    auto index = std::make_unique<Index>(200,
+                                         "test_single_batch",
+                                         "milvus_tokenizer",
+                                         "{}",
+                                         /*enable_background_merge=*/true);
     index->CreateReader(milvus::index::SetBitsetGrowing);
     index->RegisterAnalyzer("milvus_tokenizer", "{}");
 

--- a/internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp
+++ b/internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp
@@ -3212,11 +3212,16 @@ ChunkedSegmentSealedImpl::CreateTextIndex(FieldId field_id,
     std::string unique_id = GetUniqueFieldId(field_meta.get_id().get());
     if (!cfg.GetScalarIndexEnableMmap()) {
         // build text index in ram.
+        // Sealed interim index: no background merge — finish() ends with an
+        // explicit merge-all, and a racing policy merge (which finish()'s
+        // NoMergePolicy cannot cancel once started) would reintroduce the
+        // "segments could not be found in the SegmentManager" failure.
         index = std::make_unique<index::TextMatchIndex>(
             std::numeric_limits<int64_t>::max(),
             unique_id.c_str(),
             "milvus_tokenizer",
-            field_meta.get_analyzer_params().c_str());
+            field_meta.get_analyzer_params().c_str(),
+            /*enable_background_merge=*/false);
     } else {
         // build text index using mmap.
         index = std::make_unique<index::TextMatchIndex>(

--- a/internal/core/src/segcore/SegmentGrowingImpl.cpp
+++ b/internal/core/src/segcore/SegmentGrowingImpl.cpp
@@ -2375,7 +2375,8 @@ SegmentGrowingImpl::CreateTextIndex(FieldId field_id,
         200,
         unique_id.c_str(),
         "milvus_tokenizer",
-        field_meta.get_analyzer_params().c_str());
+        field_meta.get_analyzer_params().c_str(),
+        /*enable_background_merge=*/true);
     index->Commit();
     index->CreateReader(milvus::index::SetBitsetGrowing);
     index->RegisterAnalyzer("milvus_tokenizer",

--- a/internal/core/thirdparty/tantivy/tantivy-binding/include/tantivy-binding.h
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/include/tantivy-binding.h
@@ -373,7 +373,8 @@ RustResult tantivy_create_index(const char *field_name,
                                 uint32_t tantivy_index_version,
                                 uintptr_t num_threads,
                                 uintptr_t overall_memory_budget_in_bytes,
-                                bool enable_user_specified_doc_id);
+                                bool enable_user_specified_doc_id,
+                                bool enable_background_merge);
 
 RustResult tantivy_create_index_with_single_segment(const char *field_name,
                                                     TantivyDataType data_type,
@@ -556,7 +557,8 @@ RustResult tantivy_create_text_writer(const char *field_name,
                                       const char *analyzer_extra_info,
                                       uintptr_t num_threads,
                                       uintptr_t overall_memory_budget_in_bytes,
-                                      bool in_ram);
+                                      bool in_ram,
+                                      bool enable_background_merge);
 
 void tantivy_set_log_level(const char *level);
 

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_ngram_writer.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_ngram_writer.rs
@@ -49,6 +49,10 @@ impl IndexWriterWrapper {
         index.tokenizers().register(NGRAM_TOKENIZER, tokenizer);
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        // Ngram writers are only used for sealed index builds, which end with
+        // an explicit merge-all in finish(); background merges would only
+        // waste IO and race with it.
+        index_writer.set_merge_policy(Box::new(tantivy::merge_policy::NoMergePolicy));
 
         Ok(IndexWriterWrapper::V7(IndexWriterWrapperImpl {
             field,
@@ -56,6 +60,8 @@ impl IndexWriterWrapper {
             index: Arc::new(index),
             enable_user_specified_doc_id: true,
             id_field: None,
+            // Sealed-build only; merge-all runs in finish().
+            enable_background_merge: false,
         }))
     }
 }

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text.rs
@@ -126,6 +126,7 @@ mod tests {
             1,
             50_000_000,
             false,
+            false,
             TantivyIndexVersion::default_version(),
         )
         .unwrap();
@@ -164,6 +165,7 @@ mod tests {
             1,
             50_000_000,
             false,
+            false,
             TantivyIndexVersion::default_version(),
         )
         .unwrap();
@@ -193,6 +195,7 @@ mod tests {
             "",
             1,
             50_000_000,
+            false,
             false,
             TantivyIndexVersion::default_version(),
         )

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer.rs
@@ -31,6 +31,7 @@ impl IndexWriterWrapper {
         overall_memory_budget_in_bytes: usize,
         tanviy_index_version: TantivyIndexVersion,
         enable_user_specified_doc_id: bool,
+        enable_background_merge: bool,
     ) -> Result<IndexWriterWrapper> {
         init_log();
         match tanviy_index_version {
@@ -41,6 +42,7 @@ impl IndexWriterWrapper {
                     path,
                     num_threads,
                     overall_memory_budget_in_bytes,
+                    enable_background_merge,
                 )?;
                 Ok(IndexWriterWrapper::V5(writer))
             }
@@ -52,6 +54,7 @@ impl IndexWriterWrapper {
                     num_threads,
                     overall_memory_budget_in_bytes,
                     enable_user_specified_doc_id,
+                    enable_background_merge,
                 )?;
                 Ok(IndexWriterWrapper::V7(writer))
             }
@@ -236,6 +239,7 @@ mod tests {
                 50_000_000,
                 TantivyIndexVersion::V5,
                 false,
+                false,
             )
             .unwrap();
 
@@ -317,6 +321,7 @@ mod tests {
                 "",
                 1,
                 50_000_000,
+                false,
                 false,
                 TantivyIndexVersion::V5,
             )
@@ -408,6 +413,7 @@ mod tests {
                 100_000_000,
                 TantivyIndexVersion::V7,
                 enable,
+                false,
             )
             .unwrap();
 

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_c.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_c.rs
@@ -32,6 +32,7 @@ pub extern "C" fn tantivy_create_index(
     num_threads: usize,
     overall_memory_budget_in_bytes: usize,
     enable_user_specified_doc_id: bool,
+    enable_background_merge: bool,
 ) -> RustResult {
     let field_name_str = cstr_to_str!(field_name);
     let path_str = cstr_to_str!(path);
@@ -49,6 +50,7 @@ pub extern "C" fn tantivy_create_index(
         overall_memory_budget_in_bytes,
         tantivy_index_version,
         enable_user_specified_doc_id,
+        enable_background_merge,
     ) {
         Ok(wrapper) => RustResult::from_ptr(create_binding(wrapper)),
         Err(e) => RustResult::from_error(e.to_string()),

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_text.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_text.rs
@@ -15,6 +15,7 @@ impl IndexWriterWrapper {
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
         in_ram: bool,
+        enable_background_merge: bool,
         tanviy_index_version: TantivyIndexVersion,
     ) -> Result<IndexWriterWrapper> {
         match tanviy_index_version {
@@ -27,6 +28,7 @@ impl IndexWriterWrapper {
                     num_threads,
                     overall_memory_budget_in_bytes,
                     in_ram,
+                    enable_background_merge,
                 )?,
             )),
             TantivyIndexVersion::V7 => Ok(IndexWriterWrapper::V7(
@@ -39,6 +41,7 @@ impl IndexWriterWrapper {
                     num_threads,
                     overall_memory_budget_in_bytes,
                     in_ram,
+                    enable_background_merge,
                 )?,
             )),
         }

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_text_c.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_text_c.rs
@@ -19,6 +19,7 @@ pub extern "C" fn tantivy_create_text_writer(
     num_threads: usize,
     overall_memory_budget_in_bytes: usize,
     in_ram: bool,
+    enable_background_merge: bool,
 ) -> RustResult {
     init_log();
     let field_name_str = cstr_to_str!(field_name);
@@ -40,6 +41,7 @@ pub extern "C" fn tantivy_create_text_writer(
         num_threads,
         overall_memory_budget_in_bytes,
         in_ram,
+        enable_background_merge,
         tantivy_index_version,
     ) {
         Ok(wrapper) => RustResult::from_ptr(create_binding(wrapper)),

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer.rs
@@ -26,6 +26,7 @@ pub(crate) struct IndexWriterWrapperImpl {
     pub(crate) index_writer: Either<IndexWriter, SingleSegmentIndexWriter>,
     pub(crate) id_field: Option<Field>,
     pub(crate) _index: Arc<Index>,
+    pub(crate) enable_background_merge: bool,
 }
 
 #[inline]
@@ -100,10 +101,11 @@ impl IndexWriterWrapperImpl {
         path: String,
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
+        enable_background_merge: bool,
     ) -> Result<IndexWriterWrapperImpl> {
         info!(
-            "create index writer, field_name: {}, data_type: {:?}, tantivy_index_version 5",
-            field_name, data_type
+            "create index writer, field_name: {}, data_type: {:?}, tantivy_index_version 5, enable_background_merge: {}",
+            field_name, data_type, enable_background_merge
         );
         let mut schema_builder = Schema::builder();
         let field = schema_builder_add_field(&mut schema_builder, field_name, data_type);
@@ -113,11 +115,18 @@ impl IndexWriterWrapperImpl {
         let index = Index::create_in_dir(path.clone(), schema)?;
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        if !enable_background_merge {
+            // Sealed index builds end with an explicit merge-all in finish();
+            // background policy-driven merges would only waste IO and race
+            // with it, so disable them entirely for build-mode writers.
+            index_writer.set_merge_policy(Box::new(tantivy_5::merge_policy::NoMergePolicy));
+        }
         Ok(IndexWriterWrapperImpl {
             field,
             index_writer: Either::Left(index_writer),
             id_field: Some(id_field),
             _index: Arc::new(index),
+            enable_background_merge,
         })
     }
 
@@ -140,6 +149,8 @@ impl IndexWriterWrapperImpl {
             index_writer: Either::Right(index_writer),
             id_field: None,
             _index: Arc::new(index),
+            // Single-segment writer never runs finish()'s merge-all; value unused.
+            enable_background_merge: false,
         })
     }
 
@@ -284,19 +295,30 @@ impl IndexWriterWrapperImpl {
     }
 
     pub fn finish(self) -> Result<()> {
+        let enable_background_merge = self.enable_background_merge;
         match self.index_writer {
             Either::Left(mut index_writer) => {
                 index_writer.commit()?;
 
-                // merge all segments
-                let segment_ids = index_writer.index().searchable_segment_ids()?;
-                if segment_ids.len() > 1 {
-                    let _ = index_writer.merge(&segment_ids).wait();
+                if !enable_background_merge {
+                    // Build-mode writers use NoMergePolicy (set in new()), so no
+                    // background merge can race this explicit merge-all. Collapse
+                    // the auto-flushed segments into a single one. Background-merge
+                    // writers (e.g. growing segments) keep their own policy and are
+                    // not forced to a single segment here.
+                    let segment_ids = index_writer.index().searchable_segment_ids()?;
+                    if segment_ids.len() > 1 {
+                        index_writer.merge(&segment_ids).wait()?;
+                    }
                 }
 
                 index_writer.garbage_collect_files().wait()?;
 
                 index_writer.wait_merging_threads()?;
+
+                let metas = self._index.searchable_segment_metas()?;
+                let segment_ids: Vec<_> = metas.iter().map(|m| m.id().uuid_string()).collect();
+                info!("tantivy index_writer finish, segments: {:?}", segment_ids);
             }
             Either::Right(single_segment_index_writer) => {
                 single_segment_index_writer

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer_json_key_stats.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer_json_key_stats.rs
@@ -33,11 +33,17 @@ impl IndexWriterWrapperImpl {
         };
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        // Json key stats writers are only used for sealed index builds, which
+        // end with an explicit merge-all in finish(); background merges would
+        // only waste IO and race with it.
+        index_writer.set_merge_policy(Box::new(tantivy_5::merge_policy::NoMergePolicy));
         Ok(IndexWriterWrapperImpl {
             field,
             index_writer: Either::Left(index_writer),
             id_field: Some(id_field),
             _index: Arc::new(index),
+            // Sealed-build only; merge-all runs in finish().
+            enable_background_merge: false,
         })
     }
 }

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer_text.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v5/index_writer_text.rs
@@ -31,10 +31,11 @@ impl IndexWriterWrapperImpl {
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
         in_ram: bool,
+        enable_background_merge: bool,
     ) -> Result<IndexWriterWrapperImpl> {
         info!(
-            "create text index writer, field_name: {}, tantivy_index_version 5",
-            field_name
+            "create text index writer, field_name: {}, tantivy_index_version 5, enable_background_merge: {}",
+            field_name, enable_background_merge
         );
 
         let tokenizer = create_analyzer(tokenizer_params)?;
@@ -48,12 +49,18 @@ impl IndexWriterWrapperImpl {
         index.tokenizers().register(tokenizer_name, tokenizer);
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        if !enable_background_merge {
+            // Sealed text index builds end with an explicit merge-all in
+            // finish(); disable background policy merges for them.
+            index_writer.set_merge_policy(Box::new(tantivy_5::merge_policy::NoMergePolicy));
+        }
 
         Ok(IndexWriterWrapperImpl {
             field,
             index_writer: Either::Left(index_writer),
             id_field: Some(id_field),
             _index: Arc::new(index),
+            enable_background_merge,
         })
     }
 }

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer.rs
@@ -305,3 +305,71 @@ impl IndexWriterWrapperImpl {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use tantivy::Index;
+    use tempfile::TempDir;
+
+    use super::IndexWriterWrapperImpl;
+    use crate::data_type::TantivyDataType;
+
+    // tantivy's smallest per-thread arena (MEMORY_BUDGET_NUM_BYTES_MIN = 15 MB).
+    // With a single indexing thread this is tight enough that the doc count below
+    // spills into several auto-flushed segments before the finish-time commit,
+    // which is exactly the multi-segment build this test needs to exercise.
+    const MIN_MEMORY_BUDGET: usize = 15_000_000;
+    const NUM_DOCS: i64 = 1_000_000;
+
+    fn build_i64_writer(path: &str, enable_background_merge: bool) -> IndexWriterWrapperImpl {
+        IndexWriterWrapperImpl::new(
+            "number",
+            TantivyDataType::I64,
+            path.to_string(),
+            1, // single thread -> smallest arena -> forces multiple flushed segments
+            MIN_MEMORY_BUDGET,
+            false, // enable_user_specified_doc_id
+            enable_background_merge,
+        )
+        .unwrap()
+    }
+
+    /// A build-mode (enable_background_merge == false) V7 writer must collapse the
+    /// auto-flushed segments into exactly one searchable segment in finish().
+    ///
+    /// Regression guard for the finish-time merge-all (issue #51054): the V7 writer
+    /// previously shipped sealed indexes as many ~15 MB segments, and if the merge
+    /// is ever dropped again the index silently regresses to multi-segment with only
+    /// perf/logs to reveal it. The precondition assert keeps the test honest — it
+    /// proves the workload really produced >1 segment before finish() merged them.
+    #[test]
+    fn test_sealed_build_finishes_single_segment() {
+        let dir = TempDir::new().unwrap();
+        let mut writer = build_i64_writer(dir.path().to_str().unwrap(), false);
+        for i in 0..NUM_DOCS {
+            writer.add::<i64>(i, i as u32).unwrap();
+        }
+        writer.commit().unwrap();
+
+        // Precondition: the build workload genuinely auto-flushes multiple segments,
+        // so the single-segment assertion after finish() is meaningful.
+        let before = writer.index.searchable_segment_metas().unwrap();
+        assert!(
+            before.len() > 1,
+            "expected the build workload to auto-flush multiple segments, got {}",
+            before.len()
+        );
+
+        // finish() on a build-mode writer must merge them down to exactly one.
+        writer.finish().unwrap();
+
+        let index = Index::open_in_dir(dir.path()).unwrap();
+        let after = index.searchable_segment_metas().unwrap();
+        assert_eq!(
+            after.len(),
+            1,
+            "sealed build must produce exactly one tantivy segment, got {}",
+            after.len()
+        );
+    }
+}

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer.rs
@@ -98,6 +98,7 @@ pub struct IndexWriterWrapperImpl {
     pub(crate) index: Arc<Index>,
     pub(crate) id_field: Option<Field>,
     pub(crate) enable_user_specified_doc_id: bool,
+    pub(crate) enable_background_merge: bool,
 }
 
 impl IndexWriterWrapperImpl {
@@ -108,10 +109,11 @@ impl IndexWriterWrapperImpl {
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
         enable_user_specified_doc_id: bool,
+        enable_background_merge: bool,
     ) -> Result<IndexWriterWrapperImpl> {
         info!(
-            "create index writer, field_name: {}, data_type: {:?}, tantivy_index_version 7",
-            field_name, data_type
+            "create index writer, field_name: {}, data_type: {:?}, tantivy_index_version 7, enable_background_merge: {}",
+            field_name, data_type, enable_background_merge
         );
         let mut schema_builder = Schema::builder();
         let field = schema_builder_add_field(&mut schema_builder, field_name, data_type);
@@ -125,12 +127,19 @@ impl IndexWriterWrapperImpl {
         let index = Index::create_in_dir(path.clone(), schema)?;
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        if !enable_background_merge {
+            // Sealed index builds end with an explicit merge-all in finish();
+            // background policy-driven merges would only waste IO and race
+            // with it, so disable them entirely for build-mode writers.
+            index_writer.set_merge_policy(Box::new(tantivy::merge_policy::NoMergePolicy));
+        }
         Ok(IndexWriterWrapperImpl {
             field,
             index_writer,
             index: Arc::new(index),
             id_field,
             enable_user_specified_doc_id,
+            enable_background_merge,
         })
     }
 
@@ -268,7 +277,18 @@ impl IndexWriterWrapperImpl {
 
     pub fn finish(mut self) -> Result<()> {
         self.index_writer.commit()?;
-        // self.manual_merge();
+
+        if !self.enable_background_merge {
+            // Build-mode writers use NoMergePolicy (set in new()), so no
+            // background merge can race this explicit merge-all. Collapse the
+            // auto-flushed segments into a single one. Background-merge writers
+            // (e.g. growing segments) are left to their own policy and are not
+            // forced to a single segment here.
+            let segment_ids = self.index.searchable_segment_ids()?;
+            if segment_ids.len() > 1 {
+                self.index_writer.merge(&segment_ids).wait()?;
+            }
+        }
         block_on(self.index_writer.garbage_collect_files())?;
         self.index_writer.wait_merging_threads()?;
 

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer_json_key_stats.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer_json_key_stats.rs
@@ -33,12 +33,18 @@ impl IndexWriterWrapperImpl {
         };
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        // Json key stats writers are only used for sealed index builds, which
+        // end with an explicit merge-all in finish(); background merges would
+        // only waste IO and race with it.
+        index_writer.set_merge_policy(Box::new(tantivy::merge_policy::NoMergePolicy));
         Ok(IndexWriterWrapperImpl {
             field,
             index_writer,
             index: Arc::new(index),
             id_field: Some(id_field),
             enable_user_specified_doc_id: false,
+            // Sealed-build only; merge-all runs in finish().
+            enable_background_merge: false,
         })
     }
 }

--- a/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer_text.rs
+++ b/internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer_v7/index_writer_text.rs
@@ -32,10 +32,11 @@ impl IndexWriterWrapperImpl {
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
         in_ram: bool,
+        enable_background_merge: bool,
     ) -> Result<IndexWriterWrapperImpl> {
         info!(
-            "create text index writer, field_name: {}, tantivy_index_version 7",
-            field_name
+            "create text index writer, field_name: {}, tantivy_index_version 7, enable_background_merge: {}",
+            field_name, enable_background_merge
         );
 
         let analyzer = create_analyzer(analyzer_params, analyzer_extra_info)?;
@@ -49,6 +50,13 @@ impl IndexWriterWrapperImpl {
         index.tokenizers().register(analyzer_name, analyzer);
         let index_writer =
             index.writer_with_num_threads(num_threads, overall_memory_budget_in_bytes)?;
+        if !enable_background_merge {
+            // Sealed text index builds end with an explicit merge-all in
+            // finish(); disable background policy merges for them. Growing
+            // segments keep the default policy, otherwise the segment count
+            // grows unbounded across periodic commits.
+            index_writer.set_merge_policy(Box::new(tantivy::merge_policy::NoMergePolicy));
+        }
 
         Ok(IndexWriterWrapperImpl {
             field,
@@ -56,6 +64,7 @@ impl IndexWriterWrapperImpl {
             index: Arc::new(index),
             id_field: None,
             enable_user_specified_doc_id: true,
+            enable_background_merge,
         })
     }
 }

--- a/internal/core/thirdparty/tantivy/tantivy-wrapper.h
+++ b/internal/core/thirdparty/tantivy/tantivy-wrapper.h
@@ -90,7 +90,8 @@ struct TantivyIndexWrapper {
                         bool enable_user_specified_doc_id = true,
                         uintptr_t num_threads = DEFAULT_NUM_THREADS,
                         uintptr_t overall_memory_budget_in_bytes =
-                            DEFAULT_OVERALL_MEMORY_BUDGET_IN_BYTES) {
+                            DEFAULT_OVERALL_MEMORY_BUDGET_IN_BYTES,
+                        bool enable_background_merge = false) {
         RustResultWrapper res;
         if (inverted_single_semgnent) {
             AssertInfo(tantivy_index_version == 5,
@@ -106,7 +107,8 @@ struct TantivyIndexWrapper {
                                      tantivy_index_version,
                                      num_threads,
                                      overall_memory_budget_in_bytes,
-                                     enable_user_specified_doc_id));
+                                     enable_user_specified_doc_id,
+                                     enable_background_merge));
         }
         AssertInfo(res.result_->success,
                    "failed to create index: {}",
@@ -131,6 +133,10 @@ struct TantivyIndexWrapper {
     }
 
     // create index writer for text type with tokenizer.
+    // enable_background_merge: growing segments must keep tantivy's default
+    // merge policy (periodic commits would otherwise grow the segment count
+    // unbounded); sealed index builds pass false since finish() ends with an
+    // explicit merge-all.
     TantivyIndexWrapper(const char* field_name,
                         bool in_ram,
                         const char* path,
@@ -140,7 +146,8 @@ struct TantivyIndexWrapper {
                         const char* analyzer_extra_info = "",
                         uintptr_t num_threads = DEFAULT_NUM_THREADS,
                         uintptr_t overall_memory_budget_in_bytes =
-                            DEFAULT_OVERALL_MEMORY_BUDGET_IN_BYTES) {
+                            DEFAULT_OVERALL_MEMORY_BUDGET_IN_BYTES,
+                        bool enable_background_merge = false) {
         auto res = RustResultWrapper(
             tantivy_create_text_writer(field_name,
                                        path,
@@ -150,7 +157,8 @@ struct TantivyIndexWrapper {
                                        analyzer_extra_info,
                                        num_threads,
                                        overall_memory_budget_in_bytes,
-                                       in_ram));
+                                       in_ram,
+                                       enable_background_merge));
         AssertInfo(res.result_->success,
                    "failed to create text writer: {}",
                    res.result_->error);

